server.py

```diff
@@ -23,53 +28,137 @@
+                metadata.update_file_stored(filename, server)
+                metadata.close_connection()
+                return redirect(redirection_url, code=302)
+
+    return 'File Not Found', 404
+
+@app.route('/file_exists', methods=['GET'])
+def file_exists():
+    filename = request.args.get('uuid')
+    file_path = UPLOAD_FOLDER + secure_filename(filename)
+    print file_path
+    if (os.path.exists(file_path)):
+        return app.config['HOST'], 200
+    else:
+        return 'File not found', 404
 
+# Helper method for sending a file to another server
 def move_file(request):
     file_uuid = request.args.get('uuid')
     destination = request.args.get('destination')
     destination_with_endpoint = destination + '/write'
-    write_request = util.construct_put_request(destination_with_endpoint, file_uuid)
+    write_request = util.construct_post_request(destination_with_endpoint, file_uuid)
+    metadata = metadata_manager.MetadataManager()
+    metadata.update_file_stored(file_uuid, destination)
+    metadata.close_connection()
     return write_request
 
+# Transfers the file. This API call should not be open to all users.
 @app.route('/transfer', methods=['PUT'])
 def transfer():
     write_request = move_file(request)
     if write_request.status_code == 201:
         os.remove(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid))
+        metadata.delete_file_stored(request.args.get('uuid'), request.args.get('destination'))
+        metadata.close_connection()
     return write_request
 
+# Replicate the file. This API call should not be open to all users.
 @app.route('/replicate', methods=['PUT'])
 def replicate():
     write_request = move_file(request)
     return write_request
 
+# Deletes the file. This API call should not be open to all users.
+@app.route('/delete', methods=['DELETE'])
+def delete():
+    file_uuid = request.args.get('uuid')
+    file_path = UPLOAD_FOLDER + '/' + file_uuid
+    if (os.path.exists(file_path)):
+        os.remove(file_path)
+        metadata = metadata_manager.MetadataManager()
+        metadata.delete_file_stored(file_uuid, app.config['HOST'])
+        metadata.close_connection()
+        return 'Success', 200
+    return 'File not found', 404
+
+# Returns the log.
 @app.route('/logs', methods=['GET'])
 def logs():
     return 'logs'
 
+# Shuts down the server
+@app.route('/shutdown', methods=['GET'])
+def shutdow():
+    func = request.environ.get('werkzeug.server.shutdown')
+    if func is None:
+        raise RuntimeError('Not running with the Werkzeug Server')
+    func()
+    return 'Server is shutting down...', 200
+
+# Entry point for the app
 if __name__ == '__main__':
-    port = None
-    if len(sys.argv) > 1:
-        port = sys.argv[1]
-    if port == None:
-        app.run(host='0.0.0.0', debug=True)
-    else:
-        app.run(host='0.0.0.0', port=int(port))
+    # Default values
+    hostname = '0.0.0.0'
```

Comment:
It has to be `0.0.0.0` so that the server is visible to the network: (here)[http://stackoverflow.com/questions/7023052/flask-configure-dev-server-to-be-visible-across-the-network]